src/main/java/de/sandstorm_projects/telegramAlert/TelegramBot.java

```diff
@@ -80,4 +80,28 @@ void sendMessage(String msg) throws AlarmCallbackException {
             throw new AlarmCallbackException(error);
         }
     }
+
+    private HttpEntity createUrlEncodedFormEntity(String msg) throws UnsupportedEncodingException {
+        List<NameValuePair> params = new ArrayList<>(4);
+        params.add(new BasicNameValuePair("chat_id", chat));
+        params.add(new BasicNameValuePair("text", msg));
+        params.add(new BasicNameValuePair("disable_web_page_preview", "true"));
+        if (!parseMode.equals("text")) {
+            params.add(new BasicNameValuePair("parse_mode", parseMode));
+        }
+
+        return new UrlEncodedFormEntity(params, "UTF-8");
+    }
+
+    private HttpEntity createJsonStringEntity(String msg) {
+        StringBuilder params = new StringBuilder("{");
+        params.append("\"chat_id\":\"").append(chat).append("\"");
+        params.append(",\"text\":\"").append(msg).append("\"");
```

Comment:
Building JSON by hand is not a good idea, especially since the parameters (ie. `msg`) are **not escaped**.
This means that it's vulnerable to an injection (example: msg is `... " ...` => json is invalid => causes an error).

I'd recommend using a library to encode the json data. For example JSON-java ([org.json/json](https://mvnrepository.com/artifact/org.json/json) in mvn), see https://www.baeldung.com/java-org-json for examples.

Reply:
Changed to org.json library, thank you!